src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -456,5 +456,47 @@ public static int Count<T>(this ImmutableArray<T> items, Func<T, bool> predicate
 
             return count;
         }
+
+        internal static Dictionary<K, ImmutableArray<T>> ToDictionary<K, T>(this ImmutableArray<T> items, Func<T, K> keySelector, IEqualityComparer<K> comparer = null)
+        {
+            if (items.Length == 1)
```

Comment:
0 case is not very common, but does happen. I have added a special case for 0 in both implementations - for array and for bilder